### PR TITLE
Adds javadocs for Service Bus

### PR DESCRIPTION
**Pull request by conniey**
# Description

Adds javadocs for Service Bus.  API docs zip: [apidocs.zip](https://github.com/Azure/azure-sdk-for-java/files/12797689/apidocs.zip)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/README.md [hunk 1]
```diff
@@ -212,23 +222,30 @@ Consumer<ServiceBusReceivedMessageContext> processMessage = context -> {
     if (success) {
         try {
             context.complete();
-        } catch (Exception completionError) {
-            System.out.printf("Completion of the message %s failed\n", message.getMessageId());
-            completionError.printStackTrace();
+        } catch (RuntimeException error) {
+            System.out.printf("Completion of the message %s failed.%n Error: %s%n",
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/README.md:226]: learned about %n today :)

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusSenderClientJavaDocCodeSamples.java [hunk 2]
```diff
@@ -322,4 +324,35 @@ public void sendSessionMessage() {
...
+        // operation.  Users should use the callbacks on `subscribe` to understand the status of the send operation.
+        sender.sendMessage(message).subscribe(unused -> {
+        }, error -> {
+            System.err.println("Error occurred publishing batch: " + error);
+        }, () -> {
+            System.out.println("Send complete.");
+        });
+
+        // Dispose of the sender.
+        sender.close();
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusSenderClientJavaDocCodeSamples.java:355]: should there be a latch wait before calling close? or alternatively _Mono.using_ + subscribe?

**Review by anuchandy — APPROVED** [review]
thank you so much Connie!, this is great doc.

File: sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusSenderClientJavaDocCodeSamples.java (hunk 2 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusSenderClientJavaDocCodeSamples.java:355]: I wanted to show usage samples and remind users to close the sender.  Not some runnable code but I will add some comments.

Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusSenderClientJavaDocCodeSamples.java:355]: I see, sounds good

**Pull request merged**